src/krate.rs

```diff
@@ -785,6 +785,27 @@ impl Model for Crate {
 }
 
 /// Handles the `GET /crates` route.
+/// Returns a list of crates. Called in a variety of scenarios in the
+/// front end, including:
+/// - Alphabetical listing of crates
+/// - List of crates under a specific owner
+/// - Listing a user's followed crates
+///
+/// Notes:
+/// The different use cases this function covers is handled through passing
+/// in parameters in the GET request.
+///
+/// This function was has had functionality bolted on over the years, this
+/// needs to stop :). It was built like this to keep the number of database
+/// trasnactions low, though given Rust's low performance overhead, this is
```

Comment:
I would say database queries rather than database transactions here and on line 801. Everything in this function should just be requesting data from the database rather than modifying it, so there aren't any transactions that start and can be rolled back like in `krate::new`.